src/comments.js

```diff
@@ -332,7 +341,15 @@ function printTrailingComment(commentPath, print, options, parentNode) {
     // trailing comment for `2`. We can simulate the above by checking
     // if this a comment on its own line; normal trailing comments are
     // always at the end of another expression.
-    return concat([hardline, contents]);
+    if (isException) {
```

Comment:
Thanks for the review, I know those in-between comments are introducing a somehow different logic and it was hard not to make this over-complex. Hope you will find a better (and simpler) implementation 👍.